arch/x86/kvm/vmx/vmx.c

```diff
@@ -8501,10 +8524,30 @@ static __init void vmx_setup_user_return_msrs(void)
 
 	BUILD_BUG_ON(ARRAY_SIZE(vmx_uret_msrs_list) != MAX_NR_USER_RETURN_MSRS);
 
+#ifndef __PKVM_HYP__
 	for (i = 0; i < ARRAY_SIZE(vmx_uret_msrs_list); ++i)
 		kvm_add_user_return_msr(vmx_uret_msrs_list[i]);
+#else
+	for (i = 0; i < ARRAY_SIZE(vmx_uret_msrs_list); ++i) {
+		/*
+		 * TODO: As pkvm doesn't implement exception table so cannot
+		 * handle the exception generated by accessing MSR. To avoid
+		 * this, check cpu capability first before accessing a MSR which
+		 * might be not able to support. Once exception table is
+		 * supported, this can be removed.
+		 */
+		if (vmx_uret_msrs_list[i] == MSR_IA32_TSX_CTRL) {
```

Comment:
Question: are we sure all other MSRs from vmx_uret_msrs_list are always available?

Reply:
Ok, SGTM.